pkg/cmd/roachtest/cdc.go

```diff
@@ -1111,9 +1112,24 @@ func (k kafkaManager) install(ctx context.Context) {
 	k.c.Run(ctx, k.nodes, downloadScriptPath, folder)
 	if !k.c.isLocal() {
 		k.c.Run(ctx, k.nodes, `mkdir -p logs`)
-		k.c.Run(ctx, k.nodes, `sudo apt-get -q update 2>&1 > logs/apt-get-update.log`)
-		k.c.Run(ctx, k.nodes, `yes | sudo apt-get -q install openssl default-jre 2>&1 > logs/apt-get-install.log`)
+		if err := k.installJRE(ctx); err != nil {
+			k.c.t.Fatal(err)
+		}
+	}
+}
+
+func (k kafkaManager) installJRE(ctx context.Context) error {
+	retryOpts := retry.Options{
+		InitialBackoff: 1 * time.Minute,
+		MaxBackoff:     5 * time.Minute,
 	}
+	return retry.WithMaxAttempts(ctx, retryOpts, 3, func() error {
+		err := k.c.RunE(ctx, k.nodes, `sudo apt-get -q update 2>&1 > logs/apt-get-update.log`)
```

Comment:
were you aware of `repeatRunE`?

```
// repeatRunE is the same function as c.RunE but with an automatic retry loop.
func repeatRunE(
	ctx context.Context,
	t test.Test,
	c cluster.Cluster,
	node option.NodeListOption,
	operation string,
	args ...string,
) error {
	var lastError error
	for attempt, r := 0, retry.StartWithCtx(ctx, canaryRetryOptions); r.Next(); {
		if ctx.Err() != nil {
			return ctx.Err()
		}
		if t.Failed() {
			return fmt.Errorf("test has failed")
		}
		attempt++
		t.L().Printf("attempt %d - %s", attempt, operation)
		lastError = c.RunE(ctx, node, args...)
		if lastError != nil {
			t.L().Printf("error - retrying: %s", lastError)
			continue
		}
		return nil
	}
	return fmt.Errorf("all attempts failed for %s due to error: %s", operation, lastError)
}
```

would that work here instead?

Reply:
Ah, I forgot about repeateRunE.  Since this is a backport of what is on master, I think I'd prefer to merge this as-in and then follow up with a repeateRunE change. Let me know if you feel strongly otherwise.